tests/chunk.cpp

```diff
@@ -526,7 +526,9 @@ TEST(Chunk2DTest, DeepcopyReordered)
         }
     }
     ChunkYX<double> chunk2(select<DDimY, DDimX>(chunk.domain()));
-    deepcopy(chunk2, chunk);
+    ChunkSpan<double, DDomXY, Kokkos::HostSpace, std::experimental::layout_left>
+            chunk2_view(chunk2.data(), chunk.domain());
+    deepcopy(chunk2_view, chunk);
```

Comment:
I suggest we offer a mechanism to reorder the tags and compute the new layout. For example:
```cpp
Chunk<double, DDomXY, MemorySpace, stdex::layout_right> v1;
Chunk<double, DDomYX, MemorySpace, stdex::layout_left> v2(select<Y, X>(v1));
```
It could be with a free function or the slicing operator using a tagged `full_extent`.